acurl/src/request.pyx

```diff
@@ -18,40 +19,42 @@ cdef class Request:
         str url,
         tuple header_tuple,
         tuple cookie_tuple,
-        CURLSH *shared,
         tuple auth,
         bytes data,
         tuple cert,
     ):
         self.method = method
         self.url = url
         self.header_tuple = header_tuple
-        self.cookie_tuple = cookie_tuple
+        self.cookie_tuple = cookie_tuple # Tuple of byte strings
         self.auth = auth
         self.data = data
         self.cert = cert
-        self.session_cookies = self._get_session_cookies(shared)
-        
-    # NOTE: Need to fix the return type
-    cdef tuple _get_session_cookies(self, CURLSH* shared):
+        self.session_cookies = () # Tuple of byte strings
+
+    cdef void store_session_cookies(self, CURLSH* shared):
         cdef CURL* curl = curl_easy_init()
         acurl_easy_setopt_voidptr(curl, CURLOPT_SHARE, shared)
         # NOTE: Do dummy request in order to extract the cookies list
 
-        cookies = tuple(parse_cookie_string(c) for c in acurl_extract_cookielist(curl))
-        cookies = [c.format() for c in cookies if urlparse(self.url).hostname.lower() == c.domain.lower()]
-        return cookies
-
+        raw_cookies = tuple(parse_cookie_string(c) for c in acurl_extract_cookielist(curl))
+        session_cookies = tuple(c.format() for c in raw_cookies if urlparse(self.url).hostname.lower() == c.domain.lower())
+        self.session_cookies = session_cookies
 
     @property
     def headers(self):
         return dict(header.split(": ", 1) for header in self.header_tuple)
 
     @property
     def cookies(self):
-        request_cookies = tuple(session_cookie_for_url(self.url, k, v) for k, v in self.cookie_tuple)
-        session_cookies = tuple(session_cookie_for_url(self.url, k, v) for k, v in self.session_cookies)
-        # NOTE: Make sure these types are the same - format
+        request_cookies = ()
+        for c in self.cookie_tuple:
+            if isinstance(c, Cookie) or isinstance(c, _Cookie):
+                request_cookies += (c,)
```

Comment:
I've rearranged some of the code so that there is no direct passing in of the Cookie class. I think somewhere in one of the idris tests we have it creating a Cookie instance instead of passing through a bytestring or using a helper method so i'll need to update that in the codebase to adhere to this interface ^ 👍